common/src/main/scala/explore/AppMain.scala

```diff
@@ -79,7 +79,11 @@ trait AppMain extends IOApp {
     def repeatTokenRefresh(ssoURI: Uri, expiration: Instant, v: View[UserVault]): IO[Unit] =
       SSOClient
         .refreshToken[IO](ssoURI, expiration, v.set, IO.fromFuture)
-        .flatMap(u => repeatTokenRefresh(ssoURI, u.expiration, v))
+        .attempt
+        .flatMap {
+          case Right(u) => repeatTokenRefresh(ssoURI, u.map(_.expiration).getOrElse(expiration), v)
+          case Left(_)  => repeatTokenRefresh(ssoURI, expiration, v)
```

Comment:
Is the `Left` case when there's no longer a valid token?

Reply:
We need to differentiate (in another PR) when there is a regular failure which we are ignoring here and when the user was kicked out of SSO for some reason
I'll handle that case in another PR